sci-electronics/systemc/systemc-2.3.4-r1.ebuild

```diff
@@ -41,12 +40,27 @@ src_configure() {
 		--with-unix-layout
 }
 
+src_install() {
+	default
+	if use doc; then
+		if use examples; then
+			docompress -x /usr/share/doc/"${PF}"/examples
+		else
+			rm -r "${ED}"/usr/share/doc/"${PF}"/examples
+		fi
+	else
+		rm -r "${ED}"/usr/share/doc/"${PF}"
+	fi
+}
+
```

Comment:
it sounds to me the logic is over complicated, I'd simply introduce one USE=examples (instead of two USEs)... but it's your choice anyway ;-)

Btw, users can always mask the whole /usr/share/doc if they don't want docs

Reply:
> is over complicated

Yes it is over complicated, it is just a compatible workaround for older version https://github.com/gentoo/gentoo/blob/master/sci-electronics/systemc/systemc-2.3.1-r1.ebuild,